sympy/combinatorics/homomorphisms.py

```diff
@@ -417,3 +419,89 @@ def block_homomorphism(group, blocks):
     images = {g: Permutation([b[p[i]^g] for i in identity]) for g in group.generators}
     H = GroupHomomorphism(group, codomain, images)
     return H
+
+def group_isomorphism(G, H):
+    '''
+    Compute an isomorphism between 2 given groups.
+
+    Arguments:
+        G (a finite `FpGroup` or a `PermutationGroup`) -- First group
+        H (a finite `FpGroup` or a `PermutationGroup`) -- Second group
+
+    Returns:
+    If the groups are not isomorphic, `None` is returned.
+    If the groups are isomorphic, a corresponding `GroupHomomorphism` object is returned.
+
+    Summary:
+    Uses the approach suggested by Robert Tarjan to compute the isomorphism between two groups.
+    First, the set of generators are mapped with the elements of `H` and we check if the mapping induces an isomorphism.
+
+    '''
+
+    if isinstance(G, FpGroup):
+        if G.order() == S.Infinity:
+            raise NotImplementedError("Isomorphism methods are not implemented for infinite groups.")
+        G = G._to_perm_group()[0]
+    if isinstance(H, FpGroup):
+        if H.order() == S.Infinity:
+            raise NotImplementedError("Isomorphism methods are not implemented for infinite groups.")
+        H = H._to_perm_group()[0]
+
+    if not isinstance(G, PermutationGroup):
+        raise TypeError("The group must be a PermutationGroup or a finite FpGroup")
+    if not isinstance(H, PermutationGroup):
+        raise TypeError("The group must be a PermutationGroup or a finite FpGroup")
+
+    if (len(G) != len(H)) and (G.is_abelian != H.is_abelian):
```

Comment:
Should this be `G.order() != H.order()`?

Reply:
Yes, that should be changed.